Sources/Workspace/ManagedDependency.swift

```diff
@@ -14,195 +14,192 @@ import SourceControl
 import TSCBasic
 import TSCUtility
 
-/// An individual managed dependency.
-///
-/// Each dependency will have a checkout containing the sources at a
-/// particular revision, and may have an associated version.
-public class ManagedDependency {
-    /// Represents the state of the managed dependency.
-    public enum State: Equatable {
-
-        /// The dependency is a managed checkout.
-        case checkout(CheckoutState)
-
-        /// The dependency is in edited state.
-        ///
-        /// If the path is non-nil, the dependency is managed by a user and is
-        /// located at the path. In other words, this dependency is being used
-        /// for top of the tree style development.
-        case edited(AbsolutePath?)
+extension Workspace {
+    /// An individual managed dependency.
+    ///
+    /// Each dependency will have a checkout containing the sources at a
+    /// particular revision, and may have an associated version.
+    final public class ManagedDependency {
+        /// Represents the state of the managed dependency.
+        public enum State: Equatable {
+            /// The dependency is a managed checkout.
+            case checkout(CheckoutState)
+
+            /// The dependency is in edited state.
+            ///
+            /// If the path is non-nil, the dependency is managed by a user and is
+            /// located at the path. In other words, this dependency is being used
+            /// for top of the tree style development.
+            case edited(basedOn: ManagedDependency?, unmanagedPath: AbsolutePath?)
```

Comment:
Is it an error state for the `basedOn` to be `nil` here?  Just asking for my understanding...

Reply:
big picture, the whole "edited" feature needs to go away.

specifically, the basedOn can be nil in some situation unfortunately

Reply:
Yah, I think once we have explicit overrides, we can get rid of edit mode and therefore the entirety of managed dependencies.